pkg/conversion/convert.go

```diff
@@ -455,21 +455,43 @@ func convertWindupWhenToAnalyzer(windupWhen windup.When, where map[string]string
 	}
 	if windupWhen.Xmlfile != nil {
 		for _, xf := range windupWhen.Xmlfile {
-			if xf.Matches == "" {
-				// TODO handle systemid and publicid
-				continue
-			}
-			matches := strings.Replace(xf.Matches, "windup:", "", -1)
+			var xmlCond map[string]interface{}
+
 			namespaces := map[string]string{}
 			if xf.Namespace != nil {
 				for _, ns := range xf.Namespace {
 					namespaces[ns.Prefix] = ns.Uri
 				}
 			}
-			xmlCond := map[string]interface{}{
-				"xpath":      substituteWhere(where, matches),
-				"namespaces": namespaces,
+
+			if xf.Matches != "" {
+				matches := strings.Replace(xf.Matches, "windup:", "", -1)
+			    xmlCond = map[string]interface{}{
+				    "xpath":      substituteWhere(where, matches),
+				    "namespaces": namespaces,
+			    }
+			}
+
+			if xf.Publicid != "" {	// <xmlfile public-id=".*JBoss.+DTD Java EE.+5.*"/>
+				matches := strings.Replace(xf.Matches, "windup:", "", -1)
+			    xmlCond = map[string]interface{}{
+				    "xpath":      "//*[@public-id='regexp:"+substituteWhere(where, matches)+"']",
```

Comment:
xpath module in analyzer doesn't look to support regexps. Considering update https://github.com/konveyor/analyzer-lsp/blob/main/provider/internal/builtin/service_client.go#L181 with someling like:
- if the xpath starts with ```//*[@public-id='regexp:```, take the reg.expression and update xpath to ```//*[@public-id```
- search the updated xpath (all nodes with public-id attribute)
- check found nodes in loop for match regexp manually by golang

Other option could be extend xmlCond struct to store regexp there, but I'm open to all possible solution!

Reply:
I lean towards storing the regex in the xmlCond explicitly, but if the behavior of pulling it out automatically is consistently unsurprising I'd be cool with that too. 

@shawn-hurley thoughts?